src/Plugin/WebformHandler/CivicrmWebformHandler.php

```diff
@@ -146,6 +146,15 @@ public function postSave(WebformSubmissionInterface $webform_submission, $update
     $processor->postSave($webform_submission);
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function confirmForm(array &$form, FormStateInterface $form_state, WebformSubmissionInterface $webform_submission) {
+    $this->civicrm->initialize();
+    $processor = \Drupal::service('webform_civicrm.confirmform')->initialize($form_state);
```

Comment:
@rubofvil Thanks for the PR. Based on initial testing, can you please clarify if we really require this new service? How about -

- Just replace the `postSave()` with the `confirmForm()` function in this handler file? OR
- If we want to keep postSave() and doPayment() functionalities separately, can we just use the same service `webform_civicrm.postprocess` instead of adding a new one?

Reply:
@jitendrapurohit thx for the answer. The main issue is that in the hook/event postSave() the webform submission has not been saved in the db, so adding a doPayment() inside it, IF the Payment Processor redirects to external webiste, and then comes back.. webform submissions is never saved (confirmForm is nevever executed).
Hook/Event confirmForm() runs after the submissions is saved to DB, so including doPayment() here, fixes this particular issue.

- About the first proposal:

> Just replace the postSave() with the confirmForm() function in this handler file

We din't make this proposal because we think that is a big change move all development to confirmForm. Make sense, this proposal if this big change is acceptable.

- About the second proposal:

> If we want to keep postSave() and doPayment() functionalities separately, can we just use the same service webform_civicrm.postprocess instead of adding a new one?

We don't have the variables prepared/cooked in the own class WebformCivicrmPostProcess to launch the payment `$paymentProcessor->doPayment($params);`.
The var `$params` require elements from the instance, `$this`. Is for this reason that we don't moved all the method `submitIPNPayment` to confirmForm()
https://github.com/colemanw/webform_civicrm/blob/6.2.1/src/WebformCivicrmPostProcess.php#L2095